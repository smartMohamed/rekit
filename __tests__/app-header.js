import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import AppHeader from '@/components/app-header'

Enzyme.configure({ adapter: new Adapter() })

const navigations = [
  { label: 'All cameras', url: 'https://app.smartfrog.com/de-de/' },
  { label: 'My account', url: 'https://app.smartfrog.com/de-de/account/overview' },
  { label: 'Shop', url: 'https://www.smartfrog.com/de-de/shop' },
]

const languages = [
  { code: 'de', name: 'Deutsch' },
  { code: 'en', name: 'English' }
]

const props = {
  navigations: navigations,
  languages: languages,
  username: 'john.doe@lorem.com',
  activeLang: 'en',
  logout: () => {},
  changeLanguage: () => {}
}

describe('Logged-in Header component', () => {
  it('Should render component and match snapshot', () => {
    const wrapper = shallow(
    <AppHeader {...props}/>
    )
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('Should emit logout event when logout method is called', () => {
    const logout = jest.fn()
    const cmp = shallow(<AppHeader {...props}  logout={logout} />)
    
    cmp.find('.fk-app-header__logout').simulate('click')
    expect(logout).toBeCalled()
  })
})