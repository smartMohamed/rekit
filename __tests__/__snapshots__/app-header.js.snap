// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logged-in Header component Should render component and match snapshot 1`] = `"<div align=\\"start\\" justify=\\"start\\" class=\\"fk-flex\\"><div class=\\"fk-grid\\"><div class=\\"xs12 children fk-col\\"><header class=\\"fk-app-header\\"><div class=\\"fk-app-header__container\\"><h1 class=\\"fk-app-header__logo\\" pid=\\"header_button_home\\"><a href=\\"logoUrl\\" aria-label=\\"app\\"><span><svg viewBox=\\"0 0 100 100\\" class=\\"fk-icon\\"><path d=\\"M50.4 7C26.9 7 7.9 26 7.9 49.5S26.9 92 50.4 92s42.5-19 42.5-42.5S73.9 7 50.4 7zm0 80.8c-21.1 0-38.3-17.1-38.3-38.3 0-21.1 17.1-38.3 38.3-38.3 21.1 0 38.3 17.1 38.3 38.3 0 21.2-17.2 38.3-38.3 38.3z\\"></path><path d=\\"M50.4 15.5c-18.8 0-34 15.2-34 34 0 13.1 7.4 24.4 18.2 30.1-2.9-3.6-4.7-8.1-4.7-13.1C29.8 55.2 39 46 50.4 46S71 55.2 71 66.6c0 5-1.8 9.5-4.7 13.1 10.8-5.7 18.2-17 18.2-30.1-.1-18.9-15.3-34.1-34.1-34.1z\\"></path></svg></span></a></h1><nav class=\\"fk-app-header__nav\\"><div class=\\"fk-app-header__cam\\" pid=\\"header_button_all_cameras\\"><a href=\\"https://app.smartfrog.com/de-de/\\" aria-label=\\"All cameras\\"><span><svg viewBox=\\"0 0 72.6 77.7\\" class=\\"fk-icon\\"><path d=\\"M36.2 62.9c-17.5.4-32.1-13.5-32.3-31.2C4.5 13.9 19.6-.1 37.5.5c17 .6 30.5 14.2 31.2 31.2-.5 17.7-14.9 31.6-32.5 31.2zm0-58.9C20.7 3.8 7.6 16 7.3 31.7c.6 15.9 14 28.4 29.9 27.7 15.1-.6 27.4-12.6 27.8-27.7C64.8 16.2 51.9 3.8 36.2 4zm0 3.2C22.4 7 11 17.9 10.8 31.7c0 9.5 5.6 18.1 14.2 22.2-2.6-2.6-3.9-6.2-3.9-9.7.4-8.4 7.5-15.1 15.9-14.6 8 .2 14.4 6.7 14.6 14.6 0 3.4-1.3 6.7-3.4 9.2 8.2-4.1 13.3-12.5 13.3-21.7C61.6 17.9 50.1 7 36.2 7.2zm14.6 64.1v-8.2c-9.2 3.9-19.8 3.9-29 0v8.2H3.7c-1.7 0-3.2 1.3-3.2 3v2.8h71.6v-2.8c0-1.7-1.5-3-3.2-3H50.8z\\"></path></svg></span> All cameras</a><div class=\\"fk-app-header__cam__nav\\"></div></div><a href=\\"https://app.smartfrog.com/de-de/account/overview\\" class=\\"fk-app-header__account\\" pid=\\"header_button_my_account\\" aria-label=\\"My account\\"><span><svg viewBox=\\"0 0 77.2 77.2\\" class=\\"fk-icon\\"><path d=\\"M38.7 77C17.6 77 .4 59.8.4 38.7S17.6.4 38.7.4 77 17.6 77 38.7C77 59.8 59.8 77 38.7 77zm0-73.6C19.3 3.4 3.6 19.1 3.6 38.5s15.7 35.1 35.1 35.1 35.1-15.7 35.1-35.1C73.5 19.1 58 3.6 38.7 3.4zm9.7 40.4c6.9-5.4 8.2-15.5 2.8-22.4s-15.5-8.2-22.6-2.8-8.2 15.6-2.8 22.4c.9 1.1 1.7 1.9 2.8 2.8-6.7 3.4-8.6 10.1-9 18.1h3c.6-7.3 2.6-13.6 9.3-16.1 4.3 2.2 9.5 2.2 13.8 0 6.7 2.6 8.8 8.8 9.3 16.1h3c-.8-7.9-3-14.6-9.6-18.1z\\"></path></svg></span> My account</a><a class=\\"fk-app-header__btn-shop fk-btn fk-btn--small fk-btn--secondary\\" rel=\\"noopener noreferrer\\">Shop</a><div class=\\"fk-app-header__user\\"><span>john.doe@lorem.com</span></div><div class=\\"fk-app-header__logout\\"><span><svg preserveAspectRatio=\\"xMidYMid\\" viewBox=\\"0 0 11 5.5\\" class=\\"fk-icon\\"><path fill-rule=\\"evenodd\\" d=\\"M0-.002h11L5.5 5.502 0-.002z\\"></path></svg></span> </div><div class=\\"fk-app-header__lang\\">en<span><svg preserveAspectRatio=\\"xMidYMid\\" viewBox=\\"0 0 11 5.5\\" class=\\"fk-icon\\"><path fill-rule=\\"evenodd\\" d=\\"M0-.002h11L5.5 5.502 0-.002z\\"></path></svg></span><ul><li>Deutsch</li><li>English</li></ul></div></nav></div></header></div></div></div>"`;
